--- vbcc/declaration.c	2015-01-03 15:15:27.000000000 +0100
+++ vbcc/declaration.c	2015-01-05 17:37:49.000000000 +0100
@@ -3431,7 +3431,7 @@
 }
 
 /* declare a builtin function with up to two scalar arguments */
-struct Var *declare_builtin(char *name,int ztyp,int q1typ,int q1reg,int q2typ,int q2reg,int nosidefx,char *asm)
+struct Var *declare_builtin(char *name,int ztyp,int q1typ,int q1reg,int q2typ,int q2reg,int nosidefx,char *_asm)
 {
   struct struct_declaration *sd;
   struct Typ *t;
@@ -3464,9 +3464,9 @@
     t->next->flags=ztyp;
     v=add_var(name,t,EXTERN,0);
     v->flags|=BUILTIN;
-    if(asm||nosidefx){
+    if(_asm||nosidefx){
       v->fi=new_fi();
-      if(asm) v->fi->inline_asm=asm;
+      if(_asm) v->fi->inline_asm=_asm;
       if(nosidefx){
 	v->fi->call_cnt=v->fi->use_cnt=v->fi->change_cnt=0;
 	v->fi->flags=ALL_CALLS|ALL_USES|ALL_MODS|ALWAYS_RETURNS|NOSIDEFX;
--- vbcc/supp.h	2015-01-03 15:15:27.000000000 +0100
+++ vbcc/supp.h	2015-01-05 17:37:56.000000000 +0100
@@ -677,7 +677,7 @@
 extern int is_volatile_ic(struct IC *);
 extern struct case_table *calc_case_table(struct IC *,double);
 int calc_regs(struct IC *,int);
-struct Var *declare_builtin(char *name,int ztyp,int q1typ,int q1reg,int q2typ,int q2reg,int nosidefx,char *asm);
+struct Var *declare_builtin(char *name,int ztyp,int q1typ,int q1reg,int q2typ,int q2reg,int nosidefx,char *_asm);
 extern void emit_jump_table(FILE *,struct case_table *,char *,char *,int);
 extern void optimize(long, struct Var *);
 int bvcmp(bvtype *dest,bvtype *src,size_t len);
