--- gcc-2.95.3/gcc/c-lex.c	2014-11-19 22:11:17.584287819 +0000
+++ gcc-2.95.3/gcc/c-lex.c	2014-11-19 22:12:36.079142782 +0000
@@ -21,7 +21,6 @@
 #include "config.h"
 #include "system.h"
 
-#include "rtl.h"
 #include "tree.h"
 #include "input.h"
 #include "output.h"
@@ -29,6 +28,9 @@
 #include "c-tree.h"
 #include "flags.h"
 #include "c-parse.h"
+#ifndef RETURN
+#include "rtl.h"
+#endif
 #include "c-pragma.h"
 #include "toplev.h"
 #include "intl.h"
